### PR TITLE
Command to run build

### PR DESCRIPTION
**Pull request by csrwng**
Adds a command that takes a build json/yaml and runs the build to completion. It assumes that multiple clients that start the same build also use the same name for the build. If a build exists with that name, then that build is followed.

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov

File: tools/release-ci/pkg/buildrunner/runbuild.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var (
+	WaitForRunningTimeout = 10 * time.Minute
+	MaximumBuildDuration  = 4 * time.Hour
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/buildrunner/runbuild.go]: really?

File: tools/release-ci/pkg/buildrunner/runbuild.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+		buildName, err = extractExistingBuildName(result)
+		if err != nil {
+			return err
+		}
+	}
+	return followBuild(buildName)
+}
+
+func extractSuccessfulBuildName(str string) (string, error) {
+	createdBuildRegex := regexp.MustCompile("build \"([^\"]*)\" created")
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/buildrunner/runbuild.go]: can't we `-o name` to not need to parse potentially unstable human readable text?

File: tools/release-ci/pkg/buildrunner/runbuild.go [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	}
+	if !isNewOrPending(status) {
+		return nil
+	}
+	// Begin wait
+	creationTime, err := getBuildCreationTime(name)
+	if err != nil {
+		return err
+	}
+	for {
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/buildrunner/runbuild.go]: you can `time.Tick` in the `for`

File: tools/release-ci/pkg/buildrunner/runbuild.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+	cmd := exec.Command("oc", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func checkBuildSuccess(name string) error {
+	status, err := getBuildStatus(name)
+	if !isComplete(status) {
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/buildrunner/runbuild.go]: isn't this a logic error?

Comment (stevekuznetsov) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: based on when we call this func

File: tools/release-ci/pkg/buildrunner/runbuild.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+	}
+	return nil
+}
+
+func waitForComplete(name string) error {
+	creationTime, err := getBuildCreationTime(name)
+	if err != nil {
+		return err
+	}
+	for {
```
Comment (stevekuznetsov) [inline tools/release-ci/pkg/buildrunner/runbuild.go]: also this -- shouldn't it get eaten by `oc logs -f`?

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 1 above)
Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: Yes, seems like a good upper bound

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 3 above)
Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: will update

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 2 above)
Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: We can parse the Yaml/JSON and get the name from there. I guess I was being lazy? I don't want to make an extra call to the server if I don't need to.

Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: Will fix

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 4 above)
Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: No, following the build doesn't guarantee that the build will be done running. I've made that wrong assumption before in pipelines. There's all kinds of reasons why the `logs -f` command can terminate early and I really want to make sure that the build is done.

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 5 above)
Comment (csrwng) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: see above

File: tools/release-ci/pkg/buildrunner/runbuild.go (hunk 4 above)
Comment (stevekuznetsov) [reply tools/release-ci/pkg/buildrunner/runbuild.go]: ack

**Comment by stevekuznetsov** [comment]
Looks good

**Comment by csrwng** [comment]
@stevekuznetsov addressed your comments except for the one about using Tick. It made the code more complex and I didn't see much gained by it. I added the YAML parser to the vendor directory and glide decided to update the other deps. If you think we're better off not updating those, I can look into only adding the yaml parser.

**Comment by stevekuznetsov** [comment]
LGTM

**Pull request merged**